spyder/plugins/ipythonconsole/widgets/client.py

```diff
@@ -350,6 +352,53 @@ def _connect_control_signals(self):
         page_control.sig_show_find_widget_requested.connect(
             self.container.find_widget.show)
 
+    def _set_initial_cwd(self):
+        """Set initial cwd according to preferences."""
+        logger.debug("Setting initial working directory")
+        cwd_path = get_home_dir()
+
+        # This is for the first client
+        if self.id_['int_id'] == '1':
+            if self.get_conf(
+                'startup/use_project_or_home_directory',
+                section='workingdir'
+            ):
+                cwd_path = get_home_dir()
+                if self.container.active_project_path is not None:
+                    cwd_path = self.container.active_project_path

```

Comment:
Maybe instead of calling the `container/parent` for the `active_project_path` and the `get_working_directory()` some args should be passed through the constructor as `initial_active_project_path` and  `initial_working_directory`?

Reply:
Why do you think that's a better idea than the one implemented here? Just curious because I don't understand very well what would be the advantages of what you're proposing.

Reply:
Thinking about it is just that calling methods/attributes from the parent inside the child widget to get some strings seems a little bit off and increases coupling between container and child widget. Checking seems like there are other places where calling the parent/container is done so I guess is okay :+1: